cassandra/components.py

```diff
@@ -244,8 +287,22 @@ def addparam(self, key, value):
 
         self.params[key] = value
 
+    def addcapability(self, capability):
+        if capability in self.cap_tbl:
+            raise RuntimeError(f'Duplicate definition of capability {capability}.')
+        self.cap_tbl[capability] = self
+        self.results[capability] = {}
```

Comment:
If `self.results[capability]` can really be any type of data, what is the advantage of initializing it as a dictionary?